internal/controller/standalone_pgadmin/users.go

```diff
@@ -0,0 +1,286 @@
+	// as user management is only supported for pgAdmin v8 and higher.
+	pgadminVersion, err := r.reconcilePGAdminMajorVersion(ctx, podExecutor)
+	if err != nil {
+		return err
+	}
+	if pgadminVersion < 8 {
+		// If pgAdmin version is less than v8 and user management is being attempted,
+		// log a message clarifying that it is only supported for pgAdmin v8 and higher.
+		if len(pgadmin.Spec.Users) > 0 {
+			log.Info(fmt.Sprintf(
+				"The pgAdmin major version is v%v. User management is only supported for pgAdmin v8 and higher.",
+				pgadminVersion,
+			))
+		}
+		return err
+	}
+
+	return r.writePGAdminUsers(ctx, pgadmin, podExecutor)
+}
+
+// reconcilePGAdminMajorVersion execs into the pgAdmin pod and retrieves the pgAdmin major version
+func (r *PGAdminReconciler) reconcilePGAdminMajorVersion(ctx context.Context, exec Executor) (int, error) {
+	script := `
+PGADMIN_DIR=/usr/local/lib/python3.11/site-packages/pgadmin4
+cd $PGADMIN_DIR && python3 -c "import config; print(config.APP_RELEASE)"
+`
+
+	var err error
+	var stdin, stdout, stderr bytes.Buffer
+
+	err = exec(ctx, &stdin, &stdout, &stderr,
+		[]string{"bash", "-ceu", "--", script}...)
+
+	if err != nil {
+		return 0, err
+	}
+
+	return strconv.Atoi(strings.TrimSpace(stdout.String()))
+}
+
+// writePGAdminUsers takes the users in the pgAdmin spec and writes (adds or updates) their data
+// to both pgAdmin and the users.json file that is stored in the pgAdmin secret. If a user is
+// removed from the spec, its data is removed from users.json, but it is not deleted from pgAdmin.
+func (r *PGAdminReconciler) writePGAdminUsers(ctx context.Context, pgadmin *v1beta1.PGAdmin,
+	exec Executor) error {
+	log := logging.FromContext(ctx)
+
+	existingUserSecret := &corev1.Secret{ObjectMeta: naming.StandalonePGAdmin(pgadmin)}
+	err := errors.WithStack(
+		r.Client.Get(ctx, client.ObjectKeyFromObject(existingUserSecret), existingUserSecret))
+	if client.IgnoreNotFound(err) != nil {
+		return err
+	}
+
+	intentUserSecret := &corev1.Secret{ObjectMeta: naming.StandalonePGAdmin(pgadmin)}
+	intentUserSecret.SetGroupVersionKind(corev1.SchemeGroupVersion.WithKind("Secret"))
+
+	intentUserSecret.Annotations = naming.Merge(
+		pgadmin.Spec.Metadata.GetAnnotationsOrNil(),
+	)
+	intentUserSecret.Labels = naming.Merge(
+		pgadmin.Spec.Metadata.GetLabelsOrNil(),
+		map[string]string{
+			naming.LabelStandalonePGAdmin: pgadmin.Name,
+			naming.LabelRole:              naming.RolePGAdmin,
+		})
+
+	// Copy existing data if it exists, else initialize maps
+	if existingUserSecret.Data != nil {
+		intentUserSecret.Data = existingUserSecret.Data
+	} else {
+		intentUserSecret.Data = make(map[string][]byte)
+	}
+	if existingUserSecret.StringData != nil {
+		intentUserSecret.StringData = existingUserSecret.StringData
+	} else {
+		intentUserSecret.StringData = make(map[string]string)
+	}
+
+	setupScript := `
+PGADMIN_DIR=/usr/local/lib/python3.11/site-packages/pgadmin4
```

Comment:
:camping: Since we reference this path in a few places, it might be worthwhile to store it in a single variable. Perhaps in `internal/controller/standalone_pgadmin/config.go`?

Reply:
I'll set it as a pod EnvVar.

Reply:
Honestly, that's probably a bad idea. I don't think we've ever used that pattern, and it seems like if someone changed it (somehow) it would break all of our stuff.

Reply:
Okay, haven't done anything with this yet... Maybe just go with TJ's suggestion?